tests/FSharpLint.Core.Tests/Rules/Conventions/UsedUnderscorePrefixedElements.fs

```diff
@@ -0,0 +1,115 @@
+module FSharpLint.Core.Tests.Rules.Conventions.UsedUnderscorePrefixedElements
+
+open NUnit.Framework
+
+open FSharpLint.Rules
+
+[<TestFixture>]
+type TestConventionsUsedUnderscorePrefixedElements() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(UsedUnderscorePrefixedElements.rule)
+
+    [<Test>]
+    member this.``Use variable with underscore prefix``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let _random = System.Random()
+        printfn "%A" _random
+        () """
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.``Use variable with underscore prefix but not immediately after declaration``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let _random = System.Random()
+        ()
+        ()
+        _random"""
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.``Use variable with underscore prefix inside some expression``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let _random = System.Random()
+        "someString" + _random.ToString()"""
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.``Use variable with underscore prefix defined in tuple``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let (_random, x) = System.Random()
+        printfn "%A" _random
+        () """
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.``Not using variable with underscore prefix``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let _random = System.Random()
+        printfn "_random is not used here"
+        () """
+
+        Assert.IsFalse this.ErrorsExist
+        
+    [<Test>]
+    member this.``Using variable without underscore prefix``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let random = System.Random()
+        printfn "%A" random
+        ()"""
+
+        Assert.IsFalse this.ErrorsExist
+
+    [<Test>]
+    member this.``Use of underscore variable in function should not cause error``() =
+        this.Parse """
+module MyModule =
+    let MyFunc () =
+        let f _ = System.Random()"""
```

Comment:
@webwarrior-ws I don't think this snippet compiles? you cannot end a function with a let

Reply:
Added `()` at the end